llvm/docs/HowToUpdateDebugInfo.rst

```diff
@@ -91,8 +91,12 @@ misattributed to a block containing one of the instructions-to-be-merged.
 
 Examples of transformations that should follow this rule include:
 
-* Merging identical loads/stores which occur on both sides of a CFG diamond
-  (see the ``MergedLoadStoreMotion`` pass).
+* Hoisting identical instructions from all successors of a conditional branch
+  or sinking those from all different predecessors. For example, merging
```

Comment:
I feel "from all different predecessors" would be better written "from all paths to a postdominating block". The term postdominating should put the reader in mind of the sort of CFG you're describing.